tr1/output/objects: rewrite object mesh drawing

rr- opened the pull request:
#### Checklist

- [x] I have read the [coding conventions](https://github.com/LostArtefacts/TRX/blob/master/CONTRIBUTING.md#coding-conventions)
- [x] I have added a changelog entry about what my pull request accomplishes, or it is an internal change
- [x] I have added a readme entry about my new feature or OG bug fix, or it is a different change

#### Description

Rewrites meshes to reduce GPU bandwidth. Should improve a lot performance on Macs.
The reflections are behaving a bit different. I'm not sure it's a good thing.

#### Testing

Everything you can think of :D
Things that still get drawn using the old pipeline:
- UI 
- Shadows
- Lightings
- Debug mode things

rr- committed:
tr1/output/objects: rewrite object mesh drawing

walkawayy commented:
I'm getting an error on Mac as soon as the app starts with this log 😢 

```
libtrx/game/shell/main.c 24 main Game directory: /Applications/TR1X.app/Contents/Resources/
libtrx/game/input/backends/controller.c 375 M_FindController 0 controllers
libtrx/filesystem.c 434 File_Load Can't open file cfg/TR1X.json5
libtrx/config/file.c 153 ConfigFile_Read 'cfg/TR1X.json5' not loaded - default settings will apply
specific/s_shell.c 266 S_Shell_CreateWindow Trying GL backend 3.3
libtrx/gfx/context.c 115 GFX_Context_Attach Attaching to window 0x6000026a7cf0
libtrx/gfx/context.c 143 GFX_Context_Attach OpenGL vendor string:   Apple
libtrx/gfx/context.c 144 GFX_Context_Attach OpenGL renderer string: Apple M4 Pro
libtrx/gfx/context.c 145 GFX_Context_Attach OpenGL version string:  4.1 Metal - 89.3
libtrx/gfx/context.c 149 GFX_Context_Attach Shading version string: 4.10
```

aredfan commented:
I noticed a few things after a quick glance:

- [x] There's no caustics in water but this appears to be already present in develop, oops.
- [x] The glyphs are black in the console and inventory, but not in the main menu. The ammo counter and sprite on the HUD is also affected.
- [x] The skyboxes appears to be dark, for example in Colosseum room 20.

![20250403_075249_Colosseum](https://github.com/user-attachments/assets/50d9f166-37b7-43e8-a90b-581d4571b5ac)

rr- commented:
One more problem visible in develop – the fog ceased to work.

aredfan commented:
- [x] The inventory is oversized when it opens after Lara turns to gold. #1041, for reference.
- [x] If reflections are disabled before starting up the game, Lara appears to be black when she's standing on Midas' hand. The save crystals are black as well.
- [x] There appears to be an issue with Lara's right leg in cutscenes 2 and 4.

![1](https://github.com/user-attachments/assets/0c98f44e-ee2d-4543-9081-d8488b435279)

walkawayy commented:
Notice this room bounds issue in Lost Valley:

<img width="807" alt="Screenshot 2025-04-03 at 6 28 24 PM" src="https://github.com/user-attachments/assets/4d882ade-92b6-4745-9d9b-3ce0bebb980f" />

rr- commented:
Rebased to include the work from #2694.

> * [ ]  There's no caustics in water but this appears to be already present in develop, oops.
> * [ ]  The glyphs are black in the console and inventory, but not in the main menu. The ammo counter and sprite on the HUD is also affected.
> * [x]  The skyboxes appears to be dark, for example in Colosseum room 20.

Fixed the caustics and the fog. I believe the second issue is resolved as well – I wasn't able to reproduce it originally, but I found and since fixed a solid candidate that might've been the one to trigger it.

> * [ ]  The inventory is oversized when it opens after Lara turns to gold. [inventory: hardcode the passport and pickups FOV #1041](https://github.com/LostArtefacts/TRX/pull/1041), for reference.
> * [ ]  If reflections are disabled before starting up the game, Lara appears to be black when she's standing on Midas' hand. The save crystals are black as well.
> * [ ]  There appears to be an issue with Lara's right leg in cutscenes 2 and 4.

Fixed FOV changes not being respected (which was affecting the cutscenes, cinematic cameras – Midas hand, Scion piedestal – and the inventory). Fixed the reflections setting not being respected on the fly. Lara's leg is something I need help with as I fail to understand why it's missing some faces. @lahm86 could I ask for assistance?

> Notice this room bounds issue in Lost Valley:

I can't reproduce this on Linux – it seems to be a Macs-specific problem?

aredfan commented:
> Fixed the caustics and the fog. I believe the second issue is resolved as well – I wasn't able to reproduce it originally, but I found and since fixed a solid candidate that might've been the one to trigger it.

That did the trick, thank you.

I can confirm the boat's minor z fighting mentioned in 2694 is fixed here. 👍 

> > Notice this room bounds issue in Lost Valley:
> 
> I can't reproduce this on Linux – it seems to be a Macs-specific problem?

These are the steps to reproduce the issue described by walkawayy:

1. have the render mode set to window size
2. start up the game while its windowed, and smaller than your screen
3. expand the game's window without toggling it to full screen

**Edit:** This regression is from 2694, I think.

rr- committed:
tr1/output: fix the scissor test

rr- commented:
@aredfan @walkawayy fixed both the scissor issue around portals, and the leg mesh swap in the cutscenes.
